core/src/G3Timestream.cxx

```diff
@@ -149,6 +158,31 @@ template <class A> void G3Timestream::save(A &ar, unsigned v) const
 				break;
 			default:
 				log_fatal("Unknown timestream datatype %d", data_type_);
+			}
+			break;
+		case 32:
+			switch (data_type_) {
+			case TS_DOUBLE:
+				for (size_t i = 0; i < size(); i++)
+					inbuf[i] = int32_t(float(((double *)data_)[i]));
```

Comment:
Is the cast to `float` intentional? It seems like it will casue the data to be reduced to 24-bit precision, causing values in the range (2^24, 2^32) to lose their lower 8 bits unnecessarily.

Reply:
Yeah, I found some weird behavior when testing otherwise, where the decompressed data didn't match what I would have expect by simply casting double -> int32.  Maybe I'm misunderstanding something about what flac is doing under the hood.

Reply:
Ok, this is now sorted, preserving the data type through compression / decompression was the way to go.